de.fhdo.lemma.operationdsl/src/de/fhdo/lemma/operationdsl/validation/OperationDslValidator.xtend

```diff
@@ -749,4 +749,50 @@ class OperationDslValidator extends AbstractOperationDslValidator {
             ]
         }
     }
+
+    /**
+     * Check if a used by dependency is not also included in a depends on dependency
+     */
+    @Check
+    def checkDuplicatedOperationNodeDependenciesForUsedByNodes(OperationNode node) {
+        node.dependsOnNodes.forEach[dependsOnNode |
+            node.usedByNodes.forEach[usedByNode, index |
+                if (usedByNode.node == dependsOnNode.node) {
+                    error('''«node.name» may only have a dependsOn or usedBy dependency ''' +
+                        '''pointing to the same «usedByNode.node.name» node.''', node,
+                        OperationPackage.Literals::OPERATION_NODE__USED_BY_NODES, index)
+                }
+            ]
+        ]
+    }
+
+    /**
+     * Check conflicting depends on relationships between operation nodes
+     */
+    @Check
+    def checkConflictingDependenciesBetweenDependsOnOperationNodes(OperationNode node) {
+        node.dependsOnNodes.forEach[dependsOnNode |
+            val conflictingNode = dependsOnNode.node.dependsOnNodes.findFirst[n | n.node == node]
+
+            if (conflictingNode !== null)
+                error('''Cyclic dependsOn-relationship between the nodes «node.name» and ''' +
```

Comment:
Again, this error message is a bit too technical (```dependsOn```) and too verbose. In other occasions we just show the message `Cyclic dependency between x and y.` as an error message, because this text together with the error indicator is enough to communicate the actual problem to the modeler.